File: tests/uasm/sample/fibonacci/FibonacciTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class FibonacciTest : UdonSharpBehaviour
{
    public void Start()
    {
        int a = 0;
        int b = 1;
        for (int i = 0; i < 10; i++)
        {
            Debug.Log(a);
            int temp = a + b;
            a = b;
            b = temp;
        }
    }
}

File: tests/uasm/sample/fibonacci/FibonacciTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";
import type { UdonInt } from "@ootr/udon-assembly-ts/stubs/UdonTypes";

@UdonBehaviour()
export class FibonacciTest extends UdonSharpBehaviour {
  Start(): void {
    let a: UdonInt = 0 as UdonInt;
    let b: UdonInt = 1 as UdonInt;
    for (let i: UdonInt = 0 as UdonInt; i < (10 as UdonInt); i++) {
      Debug.Log(a);
      const temp: UdonInt = (a + b) as UdonInt;
      a = b;
      b = temp;
    }
  }
}
